lib/matplotlib/pyplot.py

```diff
@@ -1115,7 +1190,7 @@ def axes(arg=None, **kwargs):
         return fig.add_axes(arg, **kwargs)
 
 
-def delaxes(ax=None):
+def delaxes(ax: matplotlib.axes.Axes | None = None) -> None:
```

Comment:
See https://github.com/matplotlib/matplotlib/pull/24976#discussion_r1093784988

Short answer is that its unclear to me, too. But sphinx has trouble sometimes because of multiple classes just called `Axes`. I think the differentiating factor that actually matters is that it can properly resolve a type hint that is `Axes`, but cannot resolve it out of a union/composite type (e.g. `Axes | None` or `tuple[Figure, Axes]`)

Only one instance of using `matplotlib.axes.Axes` fully spelled out exists that is alone, and that could very well have just been me spotting that one when changing the ones that didn't work.

(`axes.Axes` also doesn't work because `axes` is a defined function in this file, and doing `maxes.Axes` with `import matplotlib.axes as maxes` I think would also generate sphinx problems, but perhaps the variety that could at least be silenced)

Reply:
Aha, I thought I had asked this before, but couldn't find it in all the comments that GitHub minimized.